pkg/api/plugin/api/config.go

```diff
@@ -0,0 +1,84 @@
+	GenevaMetricsAccount string `json:"genevaMetricsAccount,omitempty"`
+	// Geneva Metrics System (MDM) endpoint for metrics
+	GenevaMetricsEndpoint string `json:"genevaMetricsEndpoint,omitempty"`
+}
+
+// ImageConfig contains all images for the pods
+type ImageConfig struct {
+	// Format of the pull spec that is going to be
+	// used in the cluster.
+	Format                       string `json:"format,omitempty"`
+	ClusterMonitoringOperator    string `json:"clusterMonitoringOperator,omitempty"`
+	AzureControllers             string `json:"azureControllers,omitempty"`
+	PrometheusOperatorBase       string `json:"prometheusOperatorBase,omitempty"`
+	PrometheusBase               string `json:"prometheusBase,omitempty"`
+	PrometheusConfigReloaderBase string `json:"prometheusConfigReloaderBase,omitempty"`
+	ConfigReloaderBase           string `json:"configReloaderBase,omitempty"`
+	AlertManagerBase             string `json:"alertManagerBase,omitempty"`
+	NodeExporterBase             string `json:"nodeExporterBase,omitempty"`
+	GrafanaBase                  string `json:"grafanaBase,omitempty"`
+	KubeStateMetricsBase         string `json:"kubeStateMetricsBase,omitempty"`
+	KubeRbacProxyBase            string `json:"kubeRbacProxyBase,omitempty"`
+	OAuthProxyBase               string `json:"oAuthProxyBase,omitempty"`
+
+	MasterEtcd            string `json:"masterEtcd,omitempty"`
+	ControlPlane          string `json:"controlPlane,omitempty"`
+	Node                  string `json:"node,omitempty"`
+	ServiceCatalog        string `json:"serviceCatalog,omitempty"`
+	Sync                  string `json:"sync,omitempty"`
+	TemplateServiceBroker string `json:"templateServiceBroker,omitempty"`
+	Registry              string `json:"registry,omitempty"`
+	Router                string `json:"router,omitempty"`
+	RegistryConsole       string `json:"registryConsole,omitempty"`
+	AnsibleServiceBroker  string `json:"ansibleServiceBroker,omitempty"`
+	WebConsole            string `json:"webConsole,omitempty"`
+	Console               string `json:"console,omitempty"`
+	EtcdBackup            string `json:"etcdBackup,omitempty"`
+
+	// GenevaImagePullSecret defines secret used to pull private Azure images
+	GenevaImagePullSecret []byte `json:"genevaImagePullSecret,omitempty"`
+	// Geneva integration images
+	GenevaLogging string `json:"genevaLogging,omitempty"`
+	GenevaTDAgent string `json:"genevaTDAgent,omitempty"`
+	GenevaStatsd  string `json:"genevaStatsd,omitempty"`
+	MetricsBridge string `json:"metricsBridge,omitempty"`
+}
+
+// CertificateConfig contains all certificate configuration for the cluster.
+type CertificateConfig struct {
+	// geneva integration certificates
+	GenevaLogging CertKeyPair `json:"genevaLogging,omitempty"`
+	GenevaMetrics CertKeyPair `json:"genevaMetrics,omitempty"`
+}
+
+// CertKeyPair is an rsa private key and x509 certificate pair.
+type CertKeyPair struct {
+	Key  *rsa.PrivateKey   `json:"key,omitempty"`
```

Comment:
I think these needn't be pointers (need to cross-check what the internal API is doing).  Open a follow-up if not straightforward (probably so).

Reply:
https://github.com/openshift/openshift-azure/issues/986